anvil_consortium_manager/views.py

```diff
@@ -178,6 +190,123 @@ def form_valid(self, form):
+            )
+            # Re-render the page with a message.
+            return self.render_to_response(self.get_context_data(form=form))
+
+        # Account exists in ACM and is not verified
+        if (
+            models.Account.objects.filter(
+                email__iexact=email, date_verified__isnull=True
+            ).count()
+            == 1
+        ):
+            self.send_mail(email)
+            messages.add_message(
+                self.request,
+                messages.ERROR,
+                "This email is not verified, check your email for a verification link",
+            )
+
+        # Account exists in ACM and is verified
+        elif (
+            models.Account.objects.filter(
+                email__iexact=email, date_verified__isnull=False
+            ).count()
+            == 1
+        ):
+            messages.add_message(
+                self.request, messages.ERROR, "Account is already linked to this email."
+            )
+
+        else:
+            self.send_mail(email)
+            models.Account(
+                user=self.request.user,
+                email=email,
+                status=models.Account.INACTIVE_STATUS,
+                is_service_account=False,
+            ).save()
+            messages.add_message(
+                self.request,
+                messages.ERROR,
+                "To complete linking the account, check your email for a verification link",
+            )
+
+        return self.render_to_response(self.get_context_data(form=form))
+
+    def send_mail(self, email):
+        mail_subject = "Activate your account."
+        user = self.request.user
+        current_site = get_current_site(self.request)
+        message = render_to_string(
+            "anvil_consortium_manager/account_verification_email.html",
+            {
+                "user": user,
+                "domain": current_site.domain,
+                "uid": urlsafe_base64_encode(force_bytes(user.pk)),
+                "token": account_verification_token.make_token(user),
+            },
+        )
+        to_email = email
+        send_mail(mail_subject, message, None, [to_email], fail_silently=False)
+
+
+class AccountLinkVerify(RedirectView):
+    def get(self, request, uidb64, token):
+
+        try:
+            uid = force_str(urlsafe_base64_decode(uidb64))
+            acct = models.Account.objects.get(pk=uid)
+        except (TypeError, ValueError, OverflowError, models.Account.DoesNotExist):
+            acct = None
+
+        if acct is not None and account_verification_token.check_token(acct, token):
+            if acct.date_verified is not None:
+                messages.add_message(
+                    self.request, messages.SUCCESS, "The email is already verified."
+                )
+            else:
+                acct.status = models.Account.ACTIVE_STATUS
+                acct.date_verified = datetime.datetime.now()
```

Comment:
This should use `django.utils.timestamp.now()` instead of `datetime.datetime.now()`.

Reply:
changed to timezone.now()